pkg/webhook/admission/pod/metrics_aggregate_injector.go

```diff
@@ -66,13 +66,14 @@ func setMetricAggregationEnvVarsAndPorts(pod *corev1.Pod) error {
 
 			// The kserve container port/path is set as an EnvVar in the queue-proxy container
 			// so that it knows which port/path to scrape from the kserve-container.
-			pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, corev1.EnvVar{Name: constants.KServeContainerPrometheusMetricsPortEnvVarKey, Value: kserveContainerPromPort})
-			pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, corev1.EnvVar{Name: constants.KServeContainerPrometheusMetricsPathEnvVarKey, Value: kserveContainerPromPath})
+			pod.Spec.Containers[i].Env = utils.MergeEnvs(pod.Spec.Containers[i].Env, []corev1.EnvVar{
```

Comment:
Hi, thanks for sending the fix.
Can you also add a test that checks if no duplicated env is added?

Reply:
Hi @spolti, I added a couple of tests.